the-collab-lab/tcl-18-smart-shopping-list
# Cr va mark items as purshased

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓ | :sparkles: New feature     |
| ✓ | :link: Update dependencies |

## Description

As a user, I want to mark an item on my shopping list as purchased so the app can learn how often I buy different items.

## Related Issue

Fix #8 

## Acceptance Criteria


## Updates

Installed the **_date-fns_** library. 

## Changes

- The `ItemList` component was created.
- The **_date-fns_** library was used in the `ItemList` component, since it is used to manage data (as in this case, the time was handled).
- Imported firestore to `ItemList` component.
- `ItemList` was imported into the `ProductList` component.
- **Note:** At the time of uploading the changes to the library, the update of other dependencies (such as `react`, `react-dom`, and `react-scripts`) was uploaded. Although we tried to update only the **_date-nfs_** library, it was not possible and that is why you can see the others in the package.json file.

## Testing Steps / QA Criteria

1. When you download this branch, install the **_date-nfs_** library with `npm install date-fns`
2. `npm start` and in the app, add a new item or product to the list.
3. Check the item or product
4. In Firebase go to the token that corresponds to that item and in lastPurchasedDate change the date to yesterday and the same time you have right now (i.e. 24 hours before). Then click on the Update button.
5. Reload the list view, where you added the item, you will notice that the check is removed (This would be the same for tomorrow when 24 hours have passed).

### Before

![before-check](https://user-images.githubusercontent.com/60724393/106215592-4a596900-619f-11eb-991d-48468c080847.jpg)

### After

![after-check](https://user-images.githubusercontent.com/60724393/106215606-53e2d100-619f-11eb-986e-0315320e6e5a.jpg)
